tools/src/h5diff/ph5diff_main.c

```diff
@@ -279,11 +279,12 @@ h5diff_exit(int status)
             phdiff_dismiss_workers();
             MPI_Barrier(MPI_COMM_WORLD);
         }
-        MPI_Finalize();
-        status = EXIT_SUCCESS; /* Reset exit status, since some mpiexec commands generate output on failure
-                                  status */
     }
 
+    MPI_Finalize();
```

Comment:
The call to MPI_Finalize() needed to be moved outside the above block. Otherwise it wouldn't be called when run with `mpiexec -np 1` and would cause the program to abort with an abnormal termination message.